src/main/java/frc/robot/commands/AutoBalance.java

```diff
@@ -52,35 +61,53 @@ public void execute() {
 
     m_drivePower = Math.min(m_balanceController.calculate(m_currentAngle), 1);
     // Limit max power
-    m_drivePower = MathUtil.clamp(m_drivePower, -0.6, 0.6);
+    m_drivePower = MathUtil.clamp(m_drivePower, -0.5, 0.5);
 
-    double rollRateBalance = m_drive.getGyroRollPitch();
+    double rollRateBalance = m_drive.getGyroPitchRate();
     // if robot is tilting forward, stop moving so we don't overshoot the balance
     if (Math.abs(rollRateBalance) >= 0.3) {
-//      m_drivePower = Math.copySign(0.0, -m_drive.getGyroPitch().getDegrees());
+      m_balanceTimer.start();
+      m_drivePower = Math.copySign(0.15, -m_drive.getGyroPitch().getDegrees());
+    }
+
+    if (m_balanceTimer.hasElapsed(0.01)) {
+//      m_drivePower = Math.copySign(0.15, -m_drive.getGyroPitch().getDegrees());
       m_drivePower = 0.0;
-//      SmartDashboard.putNumber("Stop Balancing", 1);
+    }
+    if (m_balanceTimer.hasElapsed(0.75)) {
+      m_balanceTimer.stop();
+      m_balanceTimer.reset();
     }
 
     // Counter for checking if robot is truly balanced
-    if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE) {
+    if (Math.abs(m_error) < AutoConstants.ACCEPTABLE_BALANCE_ANGLE && !m_timer.hasElapsed(0.1)) {
       // if robot is balanced, stop moving and start a timer
       m_timer.start();
       m_drivePower = 0.0;
-    } else if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE && m_timer.get() >= 10) {
+      SmartDashboard.putBoolean("Timer Started", true);
+    }
+    if (Math.abs(m_error) < AutoConstants.ACCEPTABLE_BALANCE_ANGLE && m_timer.hasElapsed(2)) {
       // If robot has been balanced for 10 seconds, stop the command
       m_timer.stop();
       m_isLevel = true;
-    } else if (m_timer.get() > 0 && Math.abs(m_currentAngle) > AutoConstants.DESIRED_BALANCE_ANGLE) {
+      SmartDashboard.putBoolean("Timer Started", false);
+    }
+    if (Math.abs(m_error) < AutoConstants.ACCEPTABLE_BALANCE_ANGLE && m_timer.get() < 2) {
+      m_drivePower = 0.0;
+    }
+    if (m_timer.hasElapsed(0) && Math.abs(m_error) > AutoConstants.ACCEPTABLE_BALANCE_ANGLE) {
```

Comment:
These if statements should be mutually exclusive. You don't want unknowingly take multiple paths. You may need to reorder the statements as well to get the correct behavior.

Reply:
Ok, I think we should merge the PR before we resolve this, as like i said earlier this version of the command isn't being used anymore.